src/ripple/ledger/tests/PaymentSandbox_test.cpp

```diff
@@ -290,10 +291,50 @@ class PaymentSandbox_test : public beast::unit_test::suite
                 expect (pv.balanceHook (alice, gw, hugeAmt) != tinyAmt);
         }
     }
+    void testReserve()
+    {
+        testcase ("Reserve");
+        using namespace jtx;
+
+        beast::Journal dj;
+
+        auto accountFundsXRP = [&dj](
+            ReadView const& view, AccountID const& id) -> XRPAmount {
+            FreezeHandling const freezeHandling = fhZERO_IF_FROZEN;
+            return toAmount<XRPAmount> (accountHolds (
+                view, id, xrpCurrency (), xrpAccount (), freezeHandling, dj));
+        };
+
+        Env env (*this);
+
+        Account const alice ("alice");
+        env.fund (XRP(10000), alice);
+
+        auto const closeTime = dcSoTime () +
+                100 * env.closed ()->info ().closeTimeResolution;
+        env.close (closeTime);
+        ApplyViewImpl av (&*env.current (), tapNONE);
+        PaymentSandbox sb (&av);
+        {
+            // Spend a large amount and credit it back, the balance is the
+            // same, but with deferredCredits the balance will go below the reserve.
+            // If the reserve is checked before the deferredCredits are accounted for,
+            // the balance will go negative. This test case makes sure the balance stays
+            // non-negative.
+
+            accountSend (sb, xrpAccount (), alice, XRP(100000), dj);
+            accountSend (sb, alice, xrpAccount (), XRP(100000), dj);
+            auto const b1 = accountFundsXRP (sb, alice);
+            expect (b1 == beast::zero);
+        }
+
+    }
 
 public:
     void run ()
     {
+        testReserve();
+        return;
```

Comment:
Lol. FYI: this commit had already fixed that: https://github.com/ripple/rippled/pull/1705/commits/d399d580c3df5dedf9948c4259a8088ff217e71f